ProjectSidewalk/SidewalkWebpage
# Adding an undo feature to validate

Resolves #1398 

In this PR, I implemented an undo feature in the validate page. On the frontend, this meant keeping track of the last validated label in the current mission inside the panorama, and upon the press of the undo button, checking if that label had been pushed to the backend or had not. I also use a class I made called ModalUndo to keep track of undo logistics. If it had not, we simply pop it from the list of labels to be sent to the backend, but if it has, we compile and submit to the backend all the tracker, missionProgress & last label data that's happened since the last push to the backend. In either case, we delete any comments associated with the last label in the backend. 

In the backend, I wrote a new route that handles the undo functionality. There, we update the LabelValidationTable (deleting the validation), MissionTable (mission progress), CommentsTable (for any comments), ValidationTaskInteractionTable (for any new interactions), LabelTable (for label counts), and the UserStatsTable (to update user stats). 

I also updated the mobile frontend, basically just by creating a new button and piping it through the same ModalUndo class that I use to handle all undo actions (which is also done in web).

##### Before/After screenshots (if applicable)
**Before:**
<img width="1120" alt="image" src="https://github.com/ProjectSidewalk/SidewalkWebpage/assets/86600946/e8425c2f-f455-495b-b034-621893b299e6">
Simply no undo functionality
And on mobile:
<img width="377" alt="image" src="https://github.com/ProjectSidewalk/SidewalkWebpage/assets/86600946/1b95e7ac-6cbf-40f0-a76a-e632292ca4e6">

**After:** 
<img width="1120" alt="image" src="https://github.com/ProjectSidewalk/SidewalkWebpage/assets/86600946/31aa45fd-a202-44f0-926b-4c939ce99323">
Pressing the undo button gives the previous label:
<img width="1120" alt="image" src="https://github.com/ProjectSidewalk/SidewalkWebpage/assets/86600946/d22bf3ff-cfb4-439b-b4fe-ad2bd9f9cf30">

And on mobile:
<img width="377" alt="image" src="https://github.com/ProjectSidewalk/SidewalkWebpage/assets/86600946/053c3cd7-ec56-455f-ba5a-5d0d648c73d5">

<img width="377" alt="image" src="https://github.com/ProjectSidewalk/SidewalkWebpage/assets/86600946/933cb552-1272-4698-9689-deb63bcf86ca">



##### Testing instructions
1.  Open up validate.
2. Validate a label.
3. Press undo.
4. Repeat steps 3 and 4 as many times as you'd like, sometimes moving the mouse a lot so as to make sure labels get submitted to the backend.
5. At the end of the mission, verify that all the counts are correct. You can also go into the backend and print out what labels have been added & undone, and make sure this is consistent with what's been done on the front end.
6. Go to the next mission and repeat.

##### Things to check before submitting the PR
